e2e_tests/tests/cluster/test_checkpoints.py

```diff
@@ -8,11 +9,32 @@
 import yaml
 
 from determined import errors
-from determined.common import storage
+from determined.common import api, storage
+from determined.common.api import authentication, certs
 from tests import config as conf
 from tests import experiment as exp
 
 
+def wait_for_gc_to_finish(experiment_id: int) -> None:
+    certs.cli_cert = certs.default_load(conf.make_master_url())
+    authentication.cli_auth = authentication.Authentication(conf.make_master_url(), try_reauth=True)
+    saw_gc = False
+    # Don't wait longer than 5 minutes (as 600 half-seconds to improve our sampling resolution).
+    for _ in range(600):
+        r = api.get(conf.make_master_url(), "tasks").json()
+        names = [task["name"] for task in r.values()]
+        gc_name = f"Checkpoint GC (Experiment {experiment_id})"
+        if gc_name in names:
+            saw_gc = True
+        elif saw_gc:
+            # We previously saw checkpoint gc but now we don't, so it must have finished.
+            return
+        time.sleep(0.5)
```

Comment:
eh, i mean, too, we could miss seeing it, right? but then hopefully the test would just pass and we'd have teh little error message. this is definitely far from perfect, hopefully the task apis will help.

Reply:
We would have to expose the checkpoint gc state for an experiment to make this absolutely not flaky.